### PR TITLE
Incorporate changes discussed with Michael

### PR DESCRIPTION
**Pull request by WhatisRT**
Update MA terminology, rename `valueSize` to `outputSize` and some other minor changes.

### CODE REVIEW

File: shelley-ma/formal-spec/value.tex [hunk 1]
```diff
@@ -9,15 +9,13 @@ \section{Coin and Multi-Asset Tokens}
...
-are these administrative fields and calculations ever expected to operate on
-any types of tokens besides Ada, and this will continue to have the type $\Coin$.
+fees, rewards, treasury, and the proof of stake protocol.
 
 \begin{figure*}[t!]
   \emph{Abstract Types}
   %
   \begin{align*}
-    \var{aid} \in& ~\AssetID & \text{Asset IDs}
+    \var{aid} \in& ~\AssetName & \text{Asset Names}
```
Comment (michaelpj) [inline shelley-ma/formal-spec/value.tex]: change the variable name?

File: shelley-ma/formal-spec/value.tex [hunk 2]
```diff
@@ -28,17 +26,16 @@ \section{Coin and Multi-Asset Tokens}
...
   \end{equation*}
   %
   \emph{Abstract Functions and Values}
   %
   \begin{align*}
-    \mathsf{adaID} \in& ~\PolicyID & \text{Ada asset ID} \\
-    \mathsf{adaToken} \in& ~\AssetID & \text{Ada Token} \\
-    \fun{valueSize} \in& ~\Value \to \N & \text{Memory estimate for}~\Value
+    \mathsf{adaID} \in& ~\PolicyID & \text{Ada Policy ID} \\
+    \mathsf{adaToken} \in& ~\AssetName & \text{Ada Asset Name} \\
```
Comment (michaelpj) [inline shelley-ma/formal-spec/value.tex]: `adaName`?

File: shelley-ma/formal-spec/value.tex [hunk 3]
```diff
@@ -59,11 +56,11 @@ \section{Coin and Multi-Asset Tokens}
 
 \subsection*{Representing Multi-asset Types and Values}
 An \emph{Asset} comprises a set of different \emph{Asset Classes}, each of which has
-a unique identifier, $\AssetID$. We will informally refer to a pair $(\var{pid}, \var{aid})$ of a Policy ID and an Asset ID as a ``token''. The token for Ada is $(\mathsf{adaID}, \mathsf{adaToken})$.
+a unique identifier, $\AssetName$. We will informally refer to a pair $(\var{pid}, \var{aid})$ of a $\PolicyID$ and an $\AssetName$ as an $\AssetID$. The $\AssetID$ for Ada is $(\mathsf{adaID}, \mathsf{adaToken})$.
```
Comment (michaelpj) [inline shelley-ma/formal-spec/value.tex]: The definition of `AssetID` doesn't seem all that informal to me! Why not simply add it to the list of defined types?

Comment (michaelpj) [reply shelley-ma/formal-spec/value.tex]: You might want to check over other usages of "token" in what follows to see whether they need to be changed.

File: shelley-ma/formal-spec/value.tex [hunk 4]
```diff
@@ -59,11 +56,11 @@ \section{Coin and Multi-Asset Tokens}
...
 
 The set of tokens that are referred to by the underlying monetary policy represents the coinage that the asset supports.  A multi-asset value, $\Value$ is a map over zero or more assets
 to single asset values.  A single asset value is then a finite map from
```
Comment (michaelpj) [inline shelley-ma/formal-spec/value.tex]: "single asset value" is an odd phrase, which as far as I can tell is subsequently unused.

File: shelley-ma/formal-spec/value.tex [hunk 5]
```diff
@@ -77,24 +74,20 @@ \subsection*{Representing Multi-asset Types and Values}
     respects the restrictions that are imposed by the monetary policy.
     See sections \ref{sec:transactions} and \ref{sec:utxo} for details.
 
-  \item $\AssetID$ is a type used to distinguish different tokens with the same $\PolicyID$.
+  \item $\AssetName$ is a type used to distinguish different tokens with the same $\PolicyID$.
     Each $aid$ identifies a unique kind of token in $\var{pid}$.
```
Comment (michaelpj) [inline shelley-ma/formal-spec/value.tex]: change variable name

File: shelley-ma/formal-spec/value.tex (hunk 4 above)
Comment (michaelpj) [reply shelley-ma/formal-spec/value.tex]: Also, there's a variety of variants on "value" used somewhat inconsistently: "multi-asset value", "single asset value", "asset value", "value", "`Value`", maybe also "token bundle".

Comment (WhatisRT) [reply shelley-ma/formal-spec/value.tex]: Yeah, I think the whole paragraph just has to go, and the part about Asset Classes needs a rewrite as well.

File: shelley-ma/formal-spec/value.tex (hunk 3 above)
Comment (WhatisRT) [reply shelley-ma/formal-spec/value.tex]: We don't actually use `AssetID` anywhere, so it seems strange to define. However, I think we could go with defining `Value` as `AssetID ->0 Quantity`, which may or may not clear up some notation. I think we've had this discussion before.

File: shelley-ma/formal-spec/value.tex [hunk 6]
```diff
@@ -28,46 +26,43 @@ \section{Coin and Multi-Asset Tokens}
...
-to single asset values.  A single asset value is then a finite map from
-$\AssetID$s to quantities.
-
-For convenience, here and in the rest of the document, we
-will treat values of type $\Value$ also as non-partial functions where
-any omitted tokens in the domain are assumed to be zero.
+The set of tokens that are referred to by the underlying monetary
+policy represents the coinage that the asset supports. A multi-asset
+value, $\Value$ is a finitely supported function of two variables,
+$\PolicyID$ and $\AssetName$ to $Quantity$.
```
Comment (polinavino) [inline shelley-ma/formal-spec/value.tex]: to $\Quantity$

File: shelley-ma/formal-spec/value.tex [hunk 7]
```diff
@@ -28,46 +26,43 @@ \section{Coin and Multi-Asset Tokens}
...
   \end{align*}
   \caption{Type Definitions and auxiliary functions for Value}
   \label{fig:defs:value}
 \end{figure*}
 
 \subsection*{Representing Multi-asset Types and Values}
 An \emph{Asset} comprises a set of different \emph{Asset Classes}, each of which has
-a unique identifier, $\AssetID$. We will informally refer to a pair $(\var{pid}, \var{aid})$ of a Policy ID and an Asset ID as a ``token''. The token for Ada is $(\mathsf{adaID}, \mathsf{adaToken})$.
+a unique identifier, $\AssetName$. We will refer to a pair of a $\PolicyID$ and an
+$\AssetName$ as an $\AssetID$. The $\AssetID$ for Ada is $(\mathsf{adaID}, \mathsf{adaName})$.
```
Comment (polinavino) [inline shelley-ma/formal-spec/value.tex]: should \mathsf{adaID} be \mathsf{adaPolicy}, then?

File: shelley-ma/formal-spec/value-size.tex [hunk 8]
```diff
@@ -1,37 +1,34 @@
...
 
 In Shelley, the protocol parameter $\var{minUTxOValue}$ is used to
 disincentivize attacking nodes by putting many small outputs on the
 UTxO, permanently blocking memory. Because members of the $\Value$
-type can be arbitrary large in principle, the multi-asset ledger must
+type can be arbitrary large, the multi-asset ledger must
 accomodate by scaling the minimum amount of Ada in an output by the
-size of the output, which means that we need a function to determine
-the size of a member of $\Value$.
+size of the output.
```
Comment (polinavino) [inline shelley-ma/formal-spec/value-size.tex]: the minimum amount of Ada in an output should be scaled by the size of the *whole UTxO* - the whole "standard" UTxO, containing only coin and no datum), ie input |-> (addr, coin), 
which can vary if we decide to change the lengths of hashes or add some data to the address or whatever. This kind of change should not impact the deposit value, or at least, not implicitly.

This is why i think (now!) it makes sense to have the base `minUTxOValue` be a separate parameter that sets the deposit for the basic type of output. The extra stuff, ie Value type tokens or Datum hash, should have their additional deposit requirements estimated by a separate parameter or parameters, ie

perByteAda :: Coin 
lenDatum :: Integer
lenAssetName :: Integer
OR
perAssetIDAda :: Coin
perDatumAda :: Coin

or something like that

File: shelley-ma/formal-spec/utxo.tex [hunk 9]
```diff
@@ -142,7 +158,7 @@ \subsection*{The UTXO Transition Rule}
       \hldiff{\mathsf{adaID}~\notin \supp{\fun{forge}~tx}} \\
       ~\\
       \hldiff{\forall txout \in \txouts{txb},} \\
-      \hldiff{txout \geq \fun{coinToValue}(\fun{valueSize}~(\fun{getValue}~txout) * \fun{minUTxOValue}~pp)} \\~
+      \hldiff{txout \geq \fun{coinToValue}(\fun{outputSize}~{txout} * \fun{minUTxOValue}~pp)} \\~
```
Comment (polinavino) [inline shelley-ma/formal-spec/utxo.tex:161]: see comment below re deposit value parameters

File: shelley-ma/formal-spec/value.tex (hunk 7 above)
Comment (WhatisRT) [reply shelley-ma/formal-spec/value.tex]: Makes sense!

File: shelley-ma/formal-spec/value-size.tex (hunk 8 above)
Comment (WhatisRT) [reply shelley-ma/formal-spec/value-size.tex]: I've reworded this to take into account the UTxO keys as well. I don't really see the issue with how the deposit is currently computed, it seems to do exactly what you want.

**Review by polinavino — APPROVED** [review]

**Pull request merged**